bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -360,7 +360,7 @@ public boolean addConsumer(MqttMessageSubscriber subscriber) throws MqttExceptio
                 subscriberList = new ArrayList<>();
             }
             subscriberList.add(subscriber);
-            consumers.put(topic, subscriberList);
+            consumers.put(subscriber.getTopic(), subscriberList);
```

Comment:
The current state of the patch does not make sense to me.
Five lines above a look up is done to get the `subscriberList` and a new one is created if there is no one.
After that the subscriber is added to that list and put to the consumers using the key `subscriber.getTopic()`.
The lookup above it done using still the key `topic`.

So, if the original topic should be used, it needs to be fixed on all places.

I would also prefer to not use put if we already receive the list. We can add the element to the list without a further put operation.

Reply:
Another question is why we need to replace parts of the topic at all in this method if the key consists of the original topic...

Reply:
> So, if the original topic should be used, it needs to be fixed on all places.

I have done that now

> I would also prefer to not use put if we already receive the list. We can add the element to the list without a further put operation.

Done.

> Another question is why we need to replace parts of the topic at all in this method if the key consists of the original topic...

That's a very good question, the comment in code says something about that it is necessary for comparing it with a regex. But I think this PR is just for making the current behavior consistent. The topic replacement should be discussed in a new issue.